src/candecoder.cpp

```diff
@@ -0,0 +1,79 @@
+#include "candecoder.h"
+
+#define FIRST_CAN 0x000
+#define SECOND_CAN 0x100
+
+#define IS_FIRST_CAN true
+
+uint8_t int8Received;
+uint16_t int16Received;
+float floatReceived;
+boolean boolReceived;
+
+
+CANDecoder::CANDecoder(PinName rd, PinName td, int frequency = DEFAULT_CAN_FREQ) : CANManager(CAN_TypeDef* canPort, CAN_PINS pins, int frequency){};
+
+#if IS_FIRST_CAN
+void CANDecoder::readHandler(CAN_data msg) {
+   switch (msg.id) {
+      case (FIRST_CAN + 0):
+         int8Received = *msg.buf;
+         break;
+      case (FIRST_CAN + 1):
+         int16Received = *msg.buf;
+         break;
+      case (FIRST_CAN + 2):
+         floatReceived = *msg.buf;
```

Comment:
I think you have to cast `msg.buf` to be a float pointer. See [this line in the MainIO example](https://github.com/badgerloop-software/sc1-main-io-mbed/blob/ab4b54bf23dcb8ff2909835106d6ce68328caf5a/src/candecoder.cpp#L298).